scalafmt-core/shared/src/main/scala/org/scalafmt/internal/Router.scala

```diff
@@ -499,12 +499,19 @@ class Router(formatOps: FormatOps) {
           if leftOwner.is[Defn.ExtensionGroup] &&
             nextNonComment(formatToken).right.isNot[LeftParenOrBrace] =>
         val expireToken = getLastToken(leftOwner)
-        // Force a newline with the body of a Defn.ExtensionGroup if the
-        // method doesn't fit on a single line.
-        Seq(
-          Split(Space, 0).withSingleLine(expireToken),
-          Split(Newline, 1).withIndent(style.indent.main, expireToken, After)
-        )
+        def baseSplit(cost: Int = 0) = Split(Newline, cost)
```

Comment:
1. I'd rename it `nlSplit`
2. lets add implicit `FileLine`, so we know where the split was defined and the ordering of splits is preserved (before, Newline was after Space, and now it is before, which might change formatting in some cases).

Reply:
1. Renamed.
2. Added the implicit `FileLine`

Thanks for reviewing!